src/pages/questionPage.js

```diff
@@ -14,15 +14,22 @@ export const initQuestionPage = (data) => {
 
   const handleAnswer = (currentQuestion, selected) => {
     currentQuestion.selected = selected;
-    if (currentQuestion.correct === currentQuestion.selected) {
+    if (currentQuestion.correct === selected) {
       data.score++; //-y- if the user choose right option score increases
     }
     view.showAnswer(currentQuestion, data.score); //-y- data.score sended to showAnswer function
   };
 
   const currentQuestion = data.questions[data.currentQuestionIndex];
 
-  const props = { currentQuestion, onNextClick, handleAnswer, data }; //-y- data added to questionView. Because we want to reach data.score in questionView
+  const props = {
+    currentQuestion,
+    onNextClick,
+    handleAnswer,
+    score: data.score,
+    currentQuestionIndex: data.currentQuestionIndex,
+    questionLength: data.questions.length,
+  }; //-y- data added to questionView. Because we want to reach data.score in questionView
```

Comment:
I would probably destructure the properties you need from `data` first and then use them to pass to props. This is an extra step but makes the `props` object simpler.

Add some point (not necessarily now, but later) you should also remove temporary comments that have no long-term value and therefore do not need to be preserved for posterity 🙂 .

Also, consider a uniform name structure for (click) event handlers. Should they be similar to `onXXXClick` or `handleXXX` or something else? Whatever, as long as you are consistent.

```suggestion
  const { score, currentQuestionIndex, questions } = data;
  const props = {
    currentQuestion,
    onNextClick,
    handleAnswer,
    score,
    currentQuestionIndex,
    questionsLength: questions.length,
  };
```